docs/core/entity/calendar.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Calendar Entity
+sidebar_label: Calendar
+---
+
+A calendar entity is an entity that represents a set of events with a start
+and end date and/or time, helpful for automations. A calendar entity is derived from the [`homeassistant.components.calendar.CalendarEventDevice`](https://github.com/home-assistant/core/blob/dev/homeassistant/components/calendar/__init__.py).
+
+## Properties
+
+:::tip
+Properties should always only return information from memory and not do I/O (like network requests). Implement `update()` or `async_update()` to fetch data.
+:::
+
+| Name  | Type          | Default               | Description                                             |
+| ----- | ------------- | --------------------- | ------------------------------------------------------- |
+| event | CalendarEvent | `NotImplementedError` | The current or next upcoming `CalendarEvent` or `None`. |
+
+### States
+
+| Constant    | Description                                 |
+| ----------- | ------------------------------------------- |
+| `STATE_ON`  | The calendar has an active event.           |
+| `STATE_OFF` | The calendar does not have an active event. |
+
+## Methods
+
+### Get Events
+
+A calendar entity can return events that occur during a particular time range. Some notes for implementors:
+
+- The `start_date` is the lower bound and applied to the event's `end` (exclusive).
+- The `end_date` is the upper bound and applied to the event's `start` (exclusive).
+- Recurring events should be flattened and returned as individual `CalendarEvent`.
+
+```python
+class MyCalendar(CalendarEventDevice):
```

Comment:
I'd like to rename `CalendarEventDevice` to just `CalendarEntity`.

Reply:
Updated.